pkg/sql/backfill/mvcc_index_merger.go

```diff
@@ -205,19 +206,29 @@ func (ibm *IndexBackfillMerger) Merge(
 	var nextStart roachpb.Key
 	err := ibm.flowCtx.Cfg.DB.Txn(ctx, func(ctx context.Context, txn *kv.Txn) error {
 		// For now just grab all of the destination KVs and merge the corresponding entries.
-		kvs, err := txn.Scan(ctx, startKey, endKey, chunkSize)
+		var b roachpb.BatchRequest
+		b.TargetBytes = chunkBytes
+		b.MaxSpanRequestKeys = chunkSize
+		b.Add(&roachpb.ScanRequest{
+			RequestHeader: roachpb.RequestHeader{
+				Key:    startKey,
+				EndKey: endKey,
+			},
+			ScanFormat: roachpb.KEY_VALUES,
+		})
+		br, err := txn.Send(ctx, b)
 		if err != nil {
-			return err
+			return err.GoError()
 		}
-
-		if len(kvs) == 0 {
+		resp := br.Responses[0].GetScan()
+		if len(resp.Rows) == 0 {
 			nextStart = nil
 			return nil
 		}
 
-		destKeys := make([]roachpb.Key, len(kvs))
-		for i := range kvs {
-			sourceKV := &kvs[i]
+		destKeys := make([]roachpb.Key, len(resp.Rows))
+		for i := range resp.Rows {
+			sourceKV := &resp.Rows[i]
```

Comment:
Drive-by: it feel sort of lame to allocate a new byte slice here for each key in `destKeys`. We could amortize the allocations by using a single slice of buffer and always appending to it and then sub-slicing it.

Reply:
Done, changed implementation to use a single slice buffer.